harness/determined/common/api/experiment.py

```diff
@@ -216,12 +215,15 @@ def make_test_experiment_config(config: Dict[str, Any]) -> Dict[str, Any]:
                 "metric": config_test["searcher"]["metric"],
                 "max_length": {"batches": 1},
             },
-            "hyperparameters": generate_random_hparam_values(config.get("hyperparameters", {})),
-            "resources": {**config_test.get("resources", {"slots_per_trial": 1})},
+            "hyperparameters": generate_random_hparam_values(
+                config_test.get("hyperparameters", {})
+            ),
+            "resources": config_test.get("resources", {"slots_per_trial": 1}),
```

Comment:
can we just drop this line entirely?  I don't see its purpose anymore.

Reply:
Without this line the test will fail if the config does not already have a `resources` key. The test expects a `resources` key to be present. That was the first failing test we ran into. FWIW, I personally also like that it's clear that it will run on either the provided slots or 1 by default.

Reply:
ah, this is because most of the system depends on the master filling in default values, except the tests.

Ok. cool.